src/btree/bt_sync_obsolete.c

```diff
@@ -291,3 +290,324 @@ __wt_sync_obsolete_cleanup(WT_SESSION_IMPL *session, WT_REF *parent)
 
     return (0);
 }
+
+/*
+ * __checkpoint_cleanup_run_chk --
+ *     Check to decide if the checkpoint cleanup should continue running.
+ */
+static bool
+__checkpoint_cleanup_run_chk(WT_SESSION_IMPL *session)
+{
+    return (FLD_ISSET(S2C(session)->server_flags, WT_CONN_SERVER_CHECKPOINT_CLENAUP));
+}
+
+/*
+ * __checkpoint_cleanup_page_skip --
+ *     Return if checkpoint cleanup requires we read this page.
+ */
+static int
+__checkpoint_cleanup_page_skip(
```

Comment:
Is this a straight copy of `__sync_page_skip`? It would help with the review to know this in advance.

Reply:
Yes, it is a copy of `__sync_page_skip` with function name change.